### ISSUE TITLE
needless_borrowed_reference is too opinionated

### ISSUE THREAD

**Issue by Ralith** [rust-lang/rust-clippy#10245]
### Description

[`needless_borrowed_reference`](https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrowed_reference), newly enabled by default in 1.67 rejects patterns like `&(ref x, ref y)` and recommends the "match ergonomics" version instead. I intentionally avoid match ergonomics due to the confusion between values and references it introduces when reading code, and I'm not alone in that. By enabling this lint by default, clippy is forcing an unwanted stylistic change and making my code less readable. I would rather not have to go through every single project allowlisting this to unbreak CI, and I particularly don't want to have to do that repeatedly if more opinionated lints are liable to arise.

**Comment by romanstingler** [comment]
Maybe it would be useful to (don't hate me) introduce another lint level like *info* or *notice* which separates lints which doesn't have a bad effect but still want to inform the user/coder what would be suggested.

I don't think that it would be too bad to have a distinction like, 
hey dude this is just bad 
and 
hey dude your code is ok but in our opinion this suggestion could make it better for some reason.

**Comment by CommanderStorm** [comment]
👋🏻 Triage here.

I am are consolidating a few very closely related issues.
I think, we should track this issue in https://github.com/rust-lang/rust-clippy/issues/13616 since that talks about the same underlying pattern and has more comments on how to fix it, why and how especially given the 2024 edition.

**Comment by Ralith** [comment]
I disagree that this is a duplicate. The tentative consensus on that issue is that clippy should lint in *more* scenarios. This issue is a request for it *not* to have this unwanted stylistic lint by default at all.

**Comment by CommanderStorm** [comment]
It is an complexity lint, not style. So it lints for code that can be less complex.. I think if you remove the references like in https://github.com/quinn-rs/quinn/pull/1499/changes#diff-d16758a27000c51f01ff213b9b2dee28f802f60c0646c949042465db92084a43 that looks less complex, but we can keep this open.

**Comment by Ralith** [comment]
I disagree that the changes recommended by the lint are a decrease in complexity. They are a decrease in *syntax*, but exactly the same operation is performed, only less transparently.
